packages/kernel-6.1/1001-Makefile-add-prepare-target-for-external-modules.patch

```diff
@@ -23,29 +23,33 @@ existing solutions.
 Signed-off-by: Ben Cressey <bcressey@amazon.com>
 Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
 ---
- Makefile | 9 +++++++++
- 1 file changed, 9 insertions(+)
+ Makefile | 13 +++++++++++++
+ 1 file changed, 13 insertions(+)
 
 diff --git a/Makefile b/Makefile
-index 23390805e..346b898eb 100644
+index 23390805e..2f78ac123 100644
 --- a/Makefile
 +++ b/Makefile
-@@ -1874,6 +1874,15 @@ else # KBUILD_EXTMOD
+@@ -1874,6 +1874,19 @@ else # KBUILD_EXTMOD
  KBUILD_BUILTIN :=
  KBUILD_MODULES := 1
  
 +PHONY += modules_prepare
-+modules_prepare: tools/objtool
++modules_prepare:
 +	$(Q)$(MAKE) $(build)=scripts/basic
 +	$(Q)$(MAKE) $(build)=scripts/dtc
 +	$(Q)$(MAKE) $(build)=scripts/mod
 +	$(Q)$(MAKE) $(build)=scripts
 +
++ifdef CONFIG_OBJTOOL
++prepare: tools/objtool
++endif
++
```

Comment:
Does this all still work OK when cross-compiling on an x86_64 host to aarch64 target, and vice versa?

Reply:
I checked it worked for building x86_64 and aarch64 from an x86_64 build host. Will try the other way around as well. Is there concerns for this not working out with regards to ordering or with regards to choosing the right thing to do?

Without this I was not able to build the kmod for aarch64 on my x86_64 machine as it would fail when trying to build objtool (not supported for arm64 on 6.1). Either way in the long run this should converge to work as it did for 5.15/5.10 as AL has in the meantime ported their changes for objtool/livepatch to support arm64. It will land with us with the next kernel update.

I think that making this conditional on objtool support for the target is the cleanest way to go (even if the implementation might look different from what I proposed here).

Reply:
> Is there concerns for this not working out with regards to ordering or with regards to choosing the right thing to do?

I've no concerns about the approach and it seems like it should work fine. It's more that `objtool` and aarch64 have been troublesome in the past (e.g. #819), and in that case the fix had to be cleaned up when we added support for builds on an aarch64 host.

Reply:
I confirmed this is working as expected building both target architectures from x86_64, as well as aarch64 build-hosts.